Nix.gitignore

```diff
@@ -0,0 +1,2 @@
+/result
+/result-*
```

Comment:
The resulting output paths are not always at the root of the source tree:

> Build the default package from the flake in the current directory:
>
> (Source: nix3-build(1))

```suggestion
result
result-*
```

Related:

- https://github.com/github/gitignore/pull/616
- https://stackoverflow.com/questions/17888695
- https://stackoverflow.com/questions/24139478